Mesh4all/m4a-firmware
# dist/tools/zep_dispatch: Topogen updated

### Contribution description

Topgen and its makefiles are updated to generate automatically a random mac address to each node. Also was added a Makefile variable to redefine a number of nodes to generate in new topology file. The process consist in use `TOPOLOGY=[name_of_file_to_generate]` to use the topology generator (`topogen`). *note*: the file doesn't exist to the topogen works correctly, so, if you want to use again the topogen and replace the generated topology file, you have to remove it.  About mac address generation, this feature allows to use foren6 layouts. to place in correct position each node (In case that you are create an node distribution model). Topologies are parsing correctly if the mac address of some node is assigned previously the build process. 



### Testing procedure
- Open foren6
- go to the zep_dispatch tool
```
cd dist/tools/zep_dispatch
```
Run the dispatch to generate a random topology file.
```
make run TOPOLOGY=file.topo NODES=5
```
the `NODES`makefile variable specify the number of nodes to be generated. each node should be received the respect and unique random mac_address.

The expected output file should see.

```
# MAC Addresses
A := D8:04:D1:B1:D2:8A:0C:DE
B := 2C:DE:A1:D2:6A:E6:A4:EA
C := C7:D3:67:C4:56:67:8F:71
D := 4D:FA:2F:49:22:28:C6:7A
E := 2C:98:2C:FE:23:B7:5D:CE

# seed = 1663382312
# Connections
A	D	0.82	0.80


# Node	X	Y	range	color
# A	75	80	44	0x43e7c9a1
# B	31	13	40	0x0
# C	77	25	29	0x0
# D	76	88	40	0x43e7c9a1
# E	26	80	25	0x0

```

### New feature added

The new script `launch.sh` allows you load all nodes in `file.topo` automatically and run them in foren6 sniffer from zep_dispatcher tool.
Only you need to type
```
make launch
````
*Note*: this script only run in ubuntu. preferred using `gnome-terminal`setting its own options parameters. 

*Recommendation*: You could run this auto-launcher in a terminal (gnome-terminal). This will look better than running under VScode_shell or antoher kind of terminal.



### Issues/PRs references
